src/utils/useSounds.tsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from 'react';
+import gameSound from '../assets/game.mp3'; // download file from firestore storage instead
+import escapeSound from '../assets/escape.wav'; // download file from firestore storage instead
+import warningSound from '../assets/warning.wav'; // download file from firestore storage instead
+import achievementSound from '../assets/achievement.mp3'; // download file from firestore storage instead
+import teleporterSound from '../assets/teleporter.mp3'; // download file from firestore storage instead
+import exitSound from '../assets/exit.mp3'; // download file from firestore storage instead
+import selectSound from '../assets/select.mp3'; // download file from firestore storage instead
+import winSound from '../assets/win.wav'; // download file from firestore storage instead
+import cheeringSound from '../assets/cheering.wav'; // download file from firestore storage instead
+import loseSound from '../assets/lose.wav'; // download file from firestore storage instead
+
+const warningAudio = new Audio(warningSound);
+const selectAudio = new Audio(selectSound);
+const achievementAudio = new Audio(achievementSound);
+const teleporterAudio = new Audio(teleporterSound);
+const exitAudio = new Audio(exitSound);
+const winAudio = new Audio(winSound)
+const cheeringAudio = new Audio(cheeringSound);
+const loseAudio = new Audio(loseSound);
+
+// ? Do we want to include Ping here?
+
+const useSounds = () => {
+  const [musicOn, setMusicOn] = useState<boolean>(true);
+  const [soundOn, setSoundOn] = useState<boolean>(true);
+  // gameAudio can play, pause. SetGameAudio is to change tracks
+  const [gameAudio, setGameAudio] = useState<HTMLAudioElement>(new Audio(gameSound));
+
+  // Game Music soundtrack useEffects
+
+  useEffect(() => {
+    if (musicOn) gameAudio.play();
+    else gameAudio.pause();
+  }, [musicOn, gameAudio])
```

Comment:
i reckon, we remove gameAudio from dependency.
Currently this useEffect will run when musicOn is toggled.
But also whenever we load a new gameAudio. (there may be other potential events that could cause this useEffect to re-run due to gameAudio variable)

So we just isolate this toggle effect to happen when musicOn is toggled